### ISSUE TITLE
ParseWithMetaLookups propagates wrong type from derived table

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#11139]
When looking up unqualified columns in the parser, such as in:

```sql
select COL1 from (select COL1 from U) as X
```

The data type seems to get lost in the propagation of `FieldProxy` and `DataTypeProxy`. While `U.COL1` still gets assigned the proper type (e.g. `INTEGER`), `X.COL1` is `OTHER`

**Comment by lukaseder** [comment]
Another case:

```sql
select COL1, X from U join (select X from O) as A on COL1 = A.X
```

The type of `A.X` is `OTHER`

**Comment by lukaseder** [comment]
The latter is due to a very subtle bug in `ScopeStack`. I'll fix both problems in this same issue.

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.15.0 and 3.14.5 (#11140)
